go/adbc/driver/flightsql/flightsql_adbc.go

```diff
@@ -635,6 +673,9 @@ func (c *cnxn) NewStatement() (adbc.Statement, error) {
 		alloc:       c.db.alloc,
 		cl:          c.cl,
 		clientCache: c.clientCache,
+		// don't copy the headers so that calling SetOption to add more
+		// headers on the connection still propagates
```

Comment:
But conversely, that means all the statements are tied together, which I think isn't desirable - consider multiple independent statements and each of them tries to set an OpenTelemetry span, for instance.

Reply:
That was how I originally implemented it.... haha. It's only when i copied the old C++ tests that i changed it. I'll put it back :)